src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplateSupport.java

```diff
@@ -81,7 +86,8 @@ public CouchbaseDocument encodeEntity(final Object entityToEncode) {
 	}
 
 	@Override
-	public <T> T decodeEntity(String id, String source, long cas, Class<T> entityClass) {
+	public <T> T decodeEntity(String id, String source, long cas, Class<T> entityClass, TransactionGetResult txResult,
```

Comment:
is this public? If so maybe there needs to be another overload where txResult is null to not break the interface

Reply:
fixed. put back the old one to call the new one. 
In the other transaction branch I have - I have factored out the decodeEntity(), applyResult() etc. methods into a new AbstractTemplateSupport class so they aren't duplicated for reactive and non-reactive.  But I didn't want to pollute this branch with refactoring.